Project adt7420

ranechita opened the pull request:
Added iio support for adt7420 and example projects

ranechita commented:
v2: 

- fixed doxygen errors

buha commented on drivers/temperature/adt7420/iio_adt7420.c:
```diff
@@ -0,0 +1,530 @@
+	}
+	ret = adt7420_reset(desc->adt7420_dev);
+	if (ret) {
+		free(desc);
```
I think if you exit the function here you also need to call:

```
/* Free the resources allocated by adt7420_init(). */
int32_t adt7420_remove(struct adt7420_dev *dev);
```

buha commented on drivers/temperature/adt7420/iio_adt7420.c:
```diff
@@ -0,0 +1,530 @@
+
+	switch (channel->address) {
+	case 0:
+		if (adt7420_is_spi(adt7420))
```
Shouldn't this interface logic be handled in the device driver itself ? 
IIO part of the driver seems like an odd place to do this to me.
Obviously this implies some small changes to the driver itself.

buha replied:
ok, this family is a bit odd, it has different way of addressing the registers on spi and on i2c, so i guess this is fine

amiclaus reviewed (commented):
looks good overall, some comments on my side. Also, adding a *.json file should be included in the PR so the project build is performed.

ranechita commented on drivers/temperature/adt7420/iio_adt7420.c:
```diff
@@ -0,0 +1,530 @@
+		if (adt7420_is_spi(adt7420))
+			adt7420_reg_read(adt7420, ADT7320_REG_T_HIGH, &temp);
+		else {
+			uint16_t temp_msb = 0, temp_lsb = 0;
```
If the device is spi, the variables would be declared and then unused. Should they be moved at the start of the function regardless?

ranechita commented on projects/adt7420-pmdz/src/examples/dummy/dummy_example.c:
```diff
@@ -0,0 +1,134 @@
+	ret = adt7420_reset(adt7420);
+	if (ret)
+		goto error;
+	no_os_mdelay(300);
```
yes, it is the standard value for device reset. The delay was now moved in the driver implementation of adt7420_reset function

ranechita commented:
v3:

- added builds.json file for project build
- added a new commit for the driver files with macros for calculating the temperature value
- modify driver adt7420_get_temperature function to use macros 
- use same macros for calculating temperature in iio functions
- addressed nits

amiclaus commented on projects/adt7420-pmdz/src/platform/maxim/main.c:
```diff
@@ -0,0 +1,90 @@
+error:
+	return 0;
+}
+
```
nit: extra line

amiclaus commented on projects/adt7420-pmdz/src/platform/maxim/main.c:
```diff
@@ -0,0 +1,90 @@
+
+	maxim_uart_stdio(uart);
+	ret = dummy_example_main();
+	if (ret < 0)
```
`if (ret)` ?

amiclaus commented on projects/adt7420-pmdz/src/platform/maxim/main.c:
```diff
@@ -0,0 +1,90 @@
+
+#ifdef IIO_EXAMPLE
+	ret = iio_example_main ();
+	if (ret < 0)
```
this doesn't seem to be solved/ at least replied.

ranechita replied:
I must've mixed it up with the other chages and forgit about it, my bad. It's fixed now

amiclaus reviewed (commented):
some small comments that need to be addressed. also we need to check if the project build finishes successfully prior the merge.

ranechita commented:
v4:

- fixed missing/exra spaces, indentations
- addressed styling issues
- reduced sleep after device reset

amiclaus commented on projects/adt7420-pmdz/src/examples/dummy/dummy_example.c:
```diff
@@ -0,0 +1,136 @@
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+#define ADT7320_L8				NO_OS_BIT(8)
```
this does not seem to be solved. check the raw file to see the actual indentations:

![image](https://user-images.githubusercontent.com/26061529/181699930-c3372cd9-ef32-461f-9ce0-6fe17f582857.png)

ranechita replied:
You're right, visual studio seems to display the indentation a little weird

amiclaus commented on projects/adt7420-pmdz/src/examples/dummy/dummy_example.c:
```diff
@@ -0,0 +1,148 @@
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+#define ADT7320_L8			NO_OS_BIT(8)
+#define ADT7320_L16			NO_OS_BIT(16)
```
inconsistent indentation with the rest of the defines.

amiclaus commented on projects/adt7420-pmdz/src/platform/maxim/main.c:
```diff
@@ -0,0 +1,89 @@
+ *
+ * @return ret - Result of the enabled examples execution.
+*******************************************************************************/
+int main ()
```
nit: `int main()`

amiclaus commented on projects/adt7420-pmdz/src/platform/maxim/parameters.h:
```diff
@@ -0,0 +1,68 @@
+/******************************************************************************/
+
+#ifdef IIO_SUPPORT
+#define INTC_DEVICE_ID 0
```
please properly indent all these defines.

amiclaus commented on projects/adt7420-pmdz/src/examples/dummy/dummy_example.c:
```diff
@@ -0,0 +1,148 @@
+		goto error;
+	ret = adt7420_reset(adt7420);
+	if (ret)
+		goto error;
```
mem leak, we should call `adt7420_remove` starting from this conditional jump

amiclaus commented on projects/adt7420-pmdz/src/platform/maxim/main.c:
```diff
@@ -0,0 +1,89 @@
+	maxim_uart_stdio(uart);
+	ret = dummy_example_main();
+	if (ret)
+		goto error;
```
mem leak: `no_os_uart_remove` should be called starting from this conditional jump.

amiclaus reviewed (approved):
looks good, some possible memory leaks and minor indenting issues should be fixed prior the merge.

ranechita commented:
v5:

- fix indentation 
- fix mem leaks
